dashboard/app/assets/javascripts/progress.js

```diff
@@ -76,15 +76,19 @@ window.dashboard.progress = (function () {
     }
   };
 
-  progress.renderStageProgress = function (stageData, progressData, clientProgress, currentLevelId) {
+  progress.renderStageProgress = function (stageData, progressData, clientProgress, currentLevelId, puzzlePage) {
     var serverProgress = progressData.levels || {};
     var currentLevelIndex = null;
 
-    var combinedProgress = stageData.levels.map(function(level, index) {
-      if (level.id === currentLevelId) {
-        currentLevelIndex = index;
-      }
+    // Determine currentLevelIndex from the URL's /puzzle/XX value.
+    var parsedLevelIndex = window.location.pathname.match(/\/puzzle\/(\d+)/);
+    if (parsedLevelIndex && parsedLevelIndex.length > 1) {
+      currentLevelIndex = parseInt(parsedLevelIndex[1]) - 1;
+    } else {
+      return;
```

Comment:
any particular reason you changed the behavior to exit early if it couldn't find the current level? It appears that the old behavior was to pass null currentLevelIndex to `dashboard.StageProgress`, does that handle null in a reasonable way? Or do we just assume that this will never happen?